File: 프로그래머스/lv0/120863. 다항식 더하기/다항식 더하기.js
function solution(polynomial) {
    var x = 0;
    var constant = 0;
    polynomial.split(/ \+ /g).map(e => {
        if(e == 'x') {
            x++;
        }
        else if(e.includes('x')){
            x += Number(e.replace('x',""));
        }
        else{
            constant += Number(e);
        }
    })
    if(x==0) return constant.toString();
    if(x==1) x='';
    if(constant==0) return x+"x";
    return x+"x + "+constant;
}

File: 프로그래머스/lv0/120863. 다항식 더하기/다항식 더하기.java
class Solution {
    public String solution(String polynomial) {
        String answer = "";
        String[] arr = polynomial.split(" \\+ ");
        int x=0, i=0;
        for(String e: arr){
            if(e.equals("x")){
                x++;
            }else if(e.contains("x")){
                x += Integer.parseInt(e.replace("x",""));
            }else{
                i += Integer.parseInt(e);
            }
        }
        if(i==0){
            answer = x==1 ? "x" : x + "x";
        }else if(x == 0){
            answer = i + "";
        }else{
            answer = x==1 ? "x" + " + " + i : x + "x" + " + " + i;
        }
        System.out.println(arr[0]);
        return answer;
    }
}